### PR TITLE
Side Nav UI Changes

### PR DESCRIPTION
**Pull request by MrRobotjs**
This changes the side nav look and feel and changes a lot of css under the hood to avoid alignment issues on different viewports. 

# Changed
- Changed: Changed the overall look and feel of the preference side nav and main side nav

# Fixed
- Fixed: Fixed the bulk actions bar laying out in the wrong position

# Theme
- Added: --side-nav-bar-width variable to control side nav bar width
- Added: Side nav selected tab now uses --side-nav-item-active-text-color on the text instead of just being white

Main Page Changes
Settings Page Side Bar Redesign

### CODE REVIEW

**Comment by therobbiedavis** [comment]
I still need to test these all across viewports and review the changes in detail, but generally speaking I'm ok with the look of this. One change I'm not in favor of is the increased border radius on the search input. It doesn't fit with the overall UX.

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 form {
-    max-height: 38px;
+    /*max-height: 38px;*/
```
Comment (therobbiedavis) [inline UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: Unless unsure if needed elsewhere, do not comment out code just go ahead and remove it.

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss [hunk 2]
```diff
@@ -17,7 +17,7 @@ form {
 
 .typeahead-input {
     border: 1px solid transparent;
-    border-radius: 4px;
+    border-radius: 50px;
```
Comment (therobbiedavis) [inline UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: Let's revert this back to 5px as pill shape does not fit within the UX.

File: UI/Web/src/app/sidenav/_components/side-nav-item/side-nav-item.component.scss [hunk 3]
```diff
@@ -22,21 +22,21 @@
...
-    padding: 0 0 0 10px;
     width: 100%;
     height: auto;
-    min-height: 40px;
+    min-height: 50px;
     overflow: hidden;
     cursor: pointer;
-    font-size: 0.9rem;
+    font-size: 1.05rem;
+    font-family: sans-serif;
```
Comment (therobbiedavis) [inline UI/Web/src/app/sidenav/_components/side-nav-item/side-nav-item.component.scss]: Can you explain why this change?

File: UI/Web/src/theme/components/_sidenav.scss [hunk 4]
```diff
@@ -1,171 +1,236 @@
...
-        overflow-y: hidden;
-        background-color: var(--side-nav-closed-bg-color);
-        border: var(--side-nav-border-closed);
-    }
+          div {
+            min-width: unset;
+            width: unset;
+            text-overflow: ellipsis;
+            display: block;
+            font-size: 13px;
```
Comment (therobbiedavis) [inline UI/Web/src/theme/components/_sidenav.scss]: Far too small and creates a ux discrepency between the home page sidebar and the preference sidebar. Let's adjust this to a manageable size for mobile as well.

File: UI/Web/src/theme/components/_sidenav.scss [hunk 5]
```diff
@@ -1,171 +1,236 @@
...
+
+      .side-nav-item {
+        font-size: 0.9rem;
+        min-height: 30px;
+        padding-left: 12px;
+        justify-content: unset;
+
+        &.active {
+          .active-highlight {
+            width: 4px;
```
Comment (therobbiedavis) [inline UI/Web/src/theme/components/_sidenav.scss]: I'm not sold on the active-highlight changes. Maybe workshop it a bit more, but we should try to keep a consistent ux and with this change the homepage sidebar and preference sidebar would have different behaviors.

**Review by therobbiedavis — CHANGES_REQUESTED** [review]
Let some comments. Overall I'm very open to some of these changes however one thing I want to stress is we want to keep the UX across the app cohesive. So if you're changing something in one place, make sure you're investigating to make sure it does not bring any experience discrepencies as a user.

File: UI/Web/src/app/sidenav/_components/side-nav-item/side-nav-item.component.scss (hunk 3 above)
Comment (MrRobotjs) [reply UI/Web/src/app/sidenav/_components/side-nav-item/side-nav-item.component.scss]: The font size needed to be increased to match the increase of the icon size for the font family I was just testing if it looked better but after removing it now I realized it looks better without the change so I agree with that being reverted

File: UI/Web/src/theme/components/_sidenav.scss (hunk 4 above)
Comment (MrRobotjs) [reply UI/Web/src/theme/components/_sidenav.scss]: I’ll increase the font size for mobile, I took inspiration from the Plex settings side bar and thought that Kavita’s UX seems inspired by Plex so I tried to match that. I can increase the font size if needed along with mobile.

File: UI/Web/src/theme/components/_sidenav.scss (hunk 5 above)
Comment (MrRobotjs) [reply UI/Web/src/theme/components/_sidenav.scss]: A rectangle active highlight didn’t make sense to me since it was floating and not flush with the left of the sidebar so again I took inspiration from the Plex settings sidebar and thought that made more sense and also since the settings sidebar and homepage have different uses (homepage tabs are flushed to the left of the sidebar and not indented, along with having card actions)

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss (hunk 1 above)
Comment (MrRobotjs) [reply UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: Understood, removing it

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss (hunk 2 above)
Comment (MrRobotjs) [reply UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: Reverting back to 4px, thinking about adding a css variable so it can be changed in a theme for users if wanted

**Comment by majora2007** [comment]
I am taking a look at this PR now that canary has been merged into develop.

**Comment by majora2007** [comment]
Sorry I merged a bunch of changes, once you update I can take a look.

**Comment by majora2007** [comment]
Once you wrap up my comments made in discord, let's rewrite the changelog note to match the expectation aka what changed for the user in plain text. 

If we want to keep the recap of the developer notes on the PR, we can make a comment with them.

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss [hunk 6]
```diff
@@ -18,73 +13,90 @@ form {
...
-    padding: 0px 6px;
-    display: inline-block;
     overflow: hidden;
     position: relative;
     z-index: 1;
     box-sizing: border-box;
     box-shadow: none;
     cursor: text;
-    background-color: var(--input-bg-color);
+    background-color: rgb(255 255 255 / 12%);
```
Comment (majora2007) [inline UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: We need a css variable for this.

File: UI/Web/src/app/app.component.scss [hunk 7]
```diff
@@ -14,6 +15,7 @@
     scrollbar-width: thin;
     mask-image: linear-gradient(to bottom, transparent, black 0%, black 95%, transparent 100%);
     -webkit-mask-image: linear-gradient(to bottom, transparent, black 0%, black 95%, transparent 100%);
+    //margin-top: 7px;
```
Comment (majora2007) [inline UI/Web/src/app/app.component.scss:18]: Remove dead code

File: UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss [hunk 8]
```diff
@@ -18,73 +13,90 @@ form {
...
+        background-color: transparent;
 
         &:focus-visible {
-            width: calc(100vw - 180px);
+            width: 100%;
         }
 
         &:empty {
-            padding-top: 6px !important;
+            /*padding-top: 6px !important;*/
```
Comment (majora2007) [inline UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss]: Remove dead code

File: UI/Web/src/theme/components/_buttons.scss [hunk 9]
```diff
@@ -37,15 +37,28 @@
...
     border-bottom-right-radius: 0;
+  }
+}
+
+.btn-actions {
+  color: var(--btn-fa-icon-color);
+  border-radius: 6px;
+
+  &:hover {
+    background-color: rgb(255 255 255 / 18%);
```
Comment (majora2007) [inline UI/Web/src/theme/components/_buttons.scss]: Need these to be a css property

File: UI/Web/src/theme/themes/dark.scss [hunk 10]
```diff
@@ -57,365 +57,319 @@
...
+  --nav-header-bg-color: rgb(22, 27, 34);
+  /* Toasts */
+  --toast-success-bg-color: rgba(59, 158, 118, 0.9);
+  --toast-error-bg-color: #BD362F;
+  --toast-info-bg-color: #2F96B4;
+  --toast-warning-bg-color: #F89406;
+  /* Alerts */
+  --alert-text-color: #fff3cd;
+  --alert-bg-color: transparent;
+  /* Checkboxes/Switch */
```
Comment (majora2007) [inline UI/Web/src/theme/themes/dark.scss]: Please revert the indentation so it's easy to track changes. Bring the newlines back between the sections. If you're using auto formatters, please disable for this project.

**Review by majora2007 — CHANGES_REQUESTED** [review]
Some comments and I noticed the icon alignment is no longer correct. 

Otherwise, the design changes I'm okay with for merging in. 

image

File: UI/Web/src/theme/utilities/_global.scss [hunk 11]
```diff
@@ -1,5 +1,9 @@
 // Global styles for the site. Keep this as small as possible
 
+html {
+    font-size: 16px;
```
Comment (majora2007) [inline UI/Web/src/theme/utilities/_global.scss]: Make this a css variable

**Comment by majora2007** [comment]
image
The indicator is not looking like it is on the current
image

Other than this and the merge conflicts, it's good to go. If you could try and wrap this today/tomorrow, that would be great. We are very close to a release boundary, I want to get this in for shakeout before then.

**Comment by MrRobotjs** [comment]
> image The indicator is not looking like it is on the current image
> 
> Other than this and the merge conflicts, it's good to go. If you could try and wrap this today/tomorrow, that would be great. We are very close to a release boundary, I want to get this in for shakeout before then.

Fixed

**Review by majora2007 — APPROVED** [review]

**Comment by majora2007** [comment]
I found a few bugs after merging your code into my branch. I'll ping you in discord if you can come and do a bit of polish afterwards.
